contracts/ligo/src/swaps/fa2_swap.mligo

```diff
@@ -0,0 +1,174 @@
+#import "../../fa2/fa2_interface.mligo" "FA2"
+
+(* ==== Types ==== *)
+
+type swap_id = nat
+
+type fa2_token =
+  [@layout:comb]
+  { token_id : FA2.token_id
+  ; amount : nat
+  }
+
+type fa2_assets =
+  [@layout:comb]
+  { fa2_address : address
+  ; tokens : fa2_token list
+  }
+
+type swap_status =
+  [@layout:comb]
+  | Open
+  | Finished of address  // keeps who has accepted the swap
+  | Cancelled
+
+type swap_offer =
+  [@layout:comb]
+  { assets_offered : fa2_assets list
+  ; assets_requested : fa2_assets list
+  }
+
+type swap_info =
+  [@layout:comb]
+  { swap_offer : swap_offer
+  ; seller : address
+  ; status : swap_status
+  }
+
+type swap_entrypoints =
+  | Start of swap_offer
+  | Cancel of swap_id
+  | Accept of swap_id
+
+type swap_storage =
+  { next_swap_id : swap_id
+  ; swaps : (swap_id, swap_info) big_map
+  }
+
+let init_storage : swap_storage =
+  { next_swap_id = 0n
+  ; swaps = (Big_map.empty : (swap_id, swap_info) big_map)
+  }
```

Comment:
Niptick, but this looks a bout out of place in the "Types" section.

Then again it doesn't fit into any other either, so I suggest to put it in its own.